core/src/main/java/org/infinispan/container/entries/ClusteredRepeatableReadEntry.java

```diff
@@ -82,7 +79,7 @@ private EntryVersion getCurrentEntryVersion(DataContainer container, Persistence
       EntryVersion prevVersion;// on origin, the version seen is acquired without the lock, so we have to retrieve it again
       // TODO: persistence should be more orthogonal to any entry type - this should be handled in interceptor
       InternalCacheEntry ice = PersistenceUtil.loadAndStoreInDataContainer(container, persistenceManager, getKey(),
-            ctx, timeService, ignored);
+            ctx, timeService, null);
```

Comment:
@pruivo AFAICT it only ignores the store if the entry exists in the data container, flags are only checked in the interceptor. So these cases could map to:

1. entry != null && entry.isLoaded()
2. entry == null
3. entry != null && !entry.isLoaded()